### PR TITLE
refactor : Configure all promise returning functions as async functions #8368

### PR DESCRIPTION
**Pull request by Edu93Jer**
Addresses: #8368 

Change promise functions to async/await functions on these files as it was told. `Files put directly into lib and lib/classes`

### CODE REVIEW

**Comment by pgrzesik** [comment]
Thanks for submitting the PR @Edu93Jer :bow: As for review, I'd like to consult with you @medikoo before commenting further - according to the original issue description, the scope of this task is to mark functions that return promises with `async` or reconfigure it if possible to not return a promise without further refactoring of the function body. Here we have some refactoring with `await` use inside `async` functions as well, is that expected as a scope of this PR or we should just stick here to marking them as `async`?

File: lib/Serverless.js [hunk 1]
```diff
@@ -76,30 +76,23 @@ class Serverless {
...
+    this.pluginManager.setCliOptions(this.processedInput.options);
+    this.pluginManager.setCliCommands(this.processedInput.commands);
+    await this.loadEnvVariables();
+    await this.service.load(this.processedInput.options);
+    await this.pluginManager.loadAllPlugins(this.service.plugins);
+    // give the CLI the plugins and commands so that it can print out
+    // information such as options when the user enters --help
+    this.cli.setLoadedPlugins(this.pluginManager.getPlugins());
+    this.cli.setLoadedCommands(this.pluginManager.getCommands());
+    return await this.pluginManager.updateAutocompleteCacheFile();
```
Comment (medikoo) [inline lib/Serverless.js:95]: No need for `await` here

File: lib/classes/Service.js [hunk 2]
```diff
@@ -46,17 +46,18 @@ class Service {
...
-    ])
-      .then(args => this.loadServiceFileParam(...args))
-      .catch(error => {
-        if (this.serverless.cli.isHelpRequest(this.serverless.processedInput)) {
-          return null;
-        }
-        throw error;
-      });
+    try {
+      const args = await BbPromise.all([
```
Comment (medikoo) [inline lib/classes/Service.js:50]: Either we should completely refactor to native promises or leave function body as it is, this is partial convert that leaves us somewhere mid-step.

Task was simply about marking _async_ function as _async functions_, but leave their body without any changes implied.

So let's maybe revert all in body changes

**Comment by Edu93Jer** [comment]
@pgrzesik Your welcome, maybe I miss understand it, I agree with you, let's wait for the review. If there is anything to be change I'll be glad to do it.
Thanks guys and Merry Belated Christmas and Happy New Year!
Cheers.

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @Edu93Jer !

As @pgrzesik pointed, you've gone a bit beyond what the issue described

It intentionally asks simply for marking functions _async_ without implying any internal changes, to avoid diving into time-taking refactors that may takes days of back & forth before we have all sound.

So ideally if we stick precisely to what issue asks for.

From what I see it covers `lib/classes` folder only partially, is that correct? It'll be nice if we can cover whole this folder with this PR (without touching function bodies)

File: lib/classes/Service.js (hunk 2 above)
Comment (Edu93Jer) [reply lib/classes/Service.js:50]: Just to clear and start working again, all we need to do is to identify async functions and mark them as `async function ()` sorry if I am misunderstanding or causing inconvenience

Comment (medikoo) [reply lib/classes/Service.js:50]: @Edu93Jer Yes, I believe it's quite clearly explained in main issue description. If you feel some of the explanations are vague or raise questions, please comment over there. We need to be sure there are no doubts in understanding the goal of the issue

**Comment by medikoo** [comment]
@Edu93Jer I saw some changes being pushed. Is this ready for re-review? If it's the case, can you update the branch with master (and ensure that conflicts are resolved) and re-request review?

**Comment by medikoo** [comment]
@Edu93Jer due to lack of activity I'm going to close it. Still if you feel like continuing on that one, just let us know, and we will be happy to reopen

**Pull request closed without merging**